File: web/packages/fuxiaochen-types/src/changelog.ts
import type { CommonModel, ListReq } from "./common";

export interface Changelog extends CommonModel {
  version: string;
  content: string;
  date?: number;
}

export interface ChangelogListReq extends ListReq {
  version?: string;
}

export interface ChangelogListResp {
  total: number;
  lists: Changelog[] | null;
}

export interface ChangelogCreateReq {
  version: string;
  content: string;
  date?: number;
}

File: internal/model/changelog.go
package model

type Changelog struct {
	CommonModel

	Version string `gorm:"size:50;not null;comment:版本号" json:"version"`
	Content string `gorm:"type:text;not null;comment:更新内容" json:"content"`
	Date    int    `gorm:"comment:手动指定的发布日期" json:"date"`
}
